### PR TITLE
REF: enforce annotation in maybe_downcast_to_dtype

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/reshape/pivot.py [hunk 1]
```diff
@@ -174,7 +174,15 @@ def __internal_pivot_table(
...
                 and not is_integer_dtype(agged[v])
             ):
-                agged[v] = maybe_downcast_to_dtype(agged[v], data[v].dtype)
+                if isinstance(agged[v], ABCDataFrame):
+                    # exclude DataFrame case bc maybe_downcast_to_dtype expects
+                    #  ArrayLike
+                    # TODO: why does test_pivot_table_doctest_case fail if
+                    # we don't do this apparently-unnecessary setitem?
+                    agged[v] = agged[v]
+                    pass
```
Comment (jreback) [inline pandas/core/reshape/pivot.py:183]: superfluous

**Review by jreback — CHANGES_REQUESTED** [review]
no user facing change right?

File: pandas/core/reshape/pivot.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/reshape/pivot.py:183]: sure, will remove

**Comment by jbrockmendel** [comment]
> no user facing change right?

correct

File: pandas/core/reshape/pivot.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/reshape/pivot.py:183]: next pass OK?

**Review by jreback — APPROVED** [review]

**Pull request merged**